tests/test_xvfb.py

```diff
@@ -201,3 +207,28 @@ def test_fixt(fixt):
     """)
     result = testdir.runpytest()
     assert result.ret == 0
+
+
+@pytest.mark.skipif(not xauth_available, reason='no xauth')
+def test_xvfb_with_xauth(testdir):
+    original_auth = os.getenv('XAUTHORITY')
+    testdir.makeini("""
+        [pytest]
+        xvfb_xauth = True
+    """)
+    testdir.makepyfile("""
+        import os
+
+        def test_xauth():
+            print('\\nXAUTHORITY: ' + os.environ['XAUTHORITY'])
+            assert os.access(os.environ['XAUTHORITY'], os.R_OK)
+    """)
+    result = testdir.runpytest('-s')
+    # Get and parse the XAUTHORITY: line
+    authline = next(l for l in result.outlines if l.startswith('XAUTHORITY:'))
+    authfile = authline.split(' ', 1)[1]
+
+    assert result.ret == 0
+    # Make sure the authfile is deleted
+    assert not os.path.isfile(authfile)
```

Comment:
So, to be on the safe side, we would

* `os.path.isfile(...) and os.access(..., os.R_OK)` inside the test, to make sure we created a file (not a directory) and it's readable
* `not os.path.exists(...)` after the test

:satisfied: